features/253-auth-sdk-callback-in-next/api.md

```diff
@@ -0,0 +1,393 @@
+  updated_by text,
+  data jsonb,
+  PRIMARY KEY(user_id),
+  UNIQUE (user_id)
+);
+```
+
+Auth gear will also have a store `SimpleUserProfileStore` which confirms `UserProfileStore` interface:
+
+Old:
+```go=
+type UserProfileStore interface {
+  CreateUserProfile(userProfile interface{}) error
+  GetUserProfile(userID string, userProfile *interface{}) error
+}
+```
+
+New:
+```go=
+// Meta is meta data part of a user profile record
+type Meta struct {
+  ID         string                 `json:"_id"`
+  Type       string                 `json:"_type"`
+  RecordID   string                 `json:"_recordID"`
+  RecordType string                 `json:"_recordType"`
+  Access     map[string]interface{} `json:"_access"`
+  OwnerID    string                 `json:"_ownerID"`
+  CreatedAt  time.Time              `json:"_createdAt"`
+  CreatedBy  string                 `json:"_createdBy"`
+  UpdatedAt  time.Time              `json:"_updatedAt"`
+  UpdatedBy  string                 `json:"_updatedBy"`
+}
+
+// Data refers the profile info of a user,
+// like username, email, age, phone number
+type Data map[string]interface{}
+
+// UserProfile refers user profile data type
+type UserProfile struct {
+  Meta
+  Data
+}
+
+type Store interface {
+  CreateUserProfile(userID string, data Data) (UserProfile, error)
+  GetUserProfile(userID string) (UserProfile, error)
+}
+```
+
+And, use `NewAuthResponse` to generate auth response,
+
+Old:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, user skydb.Record, accessToken string) AuthResponse
+```
+
+New:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, userProfile userprofile.UserProfile, accessToken string) AuthResponse
+```
+
+# More usage example
+
+## login
+
+```javascript=
+skygear.auth.signupWithUsername(username, password).then((user) => {
+  console.log(user); // user is an user record
+  console.log(user["username"]); // username of the user
+}, (error) => {
+  ;
+});
+```
+
+## update user profile
+
+```javascript=
+var user = skygear.auth.currentUser;
+user["username"] = "new-username";
+skygear.auth.updateUser(user).then((user) => {
+  console.log(user); // updated user record
+  console.log('Username is changed to: ', user["username"]);
+  return skygear.auth.whoami();
+}, (error) => {
+  console.error(error);
+});
+```
+
+## update user profile via record gear (the error of permission denied is expected)
+
+```javascript=
+var user = skygear.auth.currentUser;
+user["username"] = "new-username";
+skygear.publicDB.save(user).then((user) => {
+  ;
+}, (error) => {
+  // permission denied
+  console.error(error);
+});
+```
+
+# access current user profile
+
+```javascript=
+console.log(skygear.auth.currentUser["username"]);
+console.log(skygear.auth.currentUser["email"]);
+console.log(skygear.auth.currentUser["gender"]);
+```
+
+# access current user auth info
+
+```javascript=
+console.log(skygear.auth.isCurrentUserVerified);
```

Comment:
As mentioned above, is it better to use `skygear.auth.isCurrentUserVerified` or `skygear.auth.isUserVerified(skygear.auth.currentUser)`?

Reply:
@chpapa We will discuss user attribute design in another PR, and https://oursky.quip.com/ewfGAfyodkb0 listed draft API design. Feel free to leave comments there. Thanks.